src/Core/Pipelines/SummaryPipelineExtensions.cs

```diff
@@ -0,0 +1,41 @@
+using Microsoft.Extensions.Logging;
+using Summary.Pipes;
+using Summary.Pipes.Filters;
+
+namespace Summary.Pipelines;
+
+/// <summary>
+///     Convenient extensions for <see cref="SummaryPipeline"/>.
+/// </summary>
+public static class SummaryPipelineExtensions
+{
+    /// <summary>
+    ///     Specifies the logger factory to use for pipes inside the pipeline.
+    /// </summary>
+    /// <remarks>
+    ///     This method should be called <i>before</i> anything else so that
+    ///     given logger factory is passed into all subsequent calls.
+    /// </remarks>
+    public static SummaryPipeline UseLoggerFactory(this SummaryPipeline self, ILoggerFactory factory) =>
+        self.Customize(options => options.UseLoggerFactory(factory));
+
+    /// <summary>
+    ///     Disables <see cref="FilterPublicMembersPipe"/> filter in the pipeline.
+    ///     This makes the generator to include private members into the parsed document.
+    /// </summary>
+    public static SummaryPipeline IncludeNonPublicMembers(this SummaryPipeline self)
```

Comment:
Would it be useful to provide the ability to specify the minimum access level to generate documentation for? For example, we may want to generate documentation for `internal` members for internal purposes, but `private` members may be redundant and should thus be excluded from the output.